arcgis-ios-sdk-samples/Utility Network/Configure subnetwork trace/ConfigureSubnetworkTraceOptionsViewController.swift

```diff
@@ -0,0 +1,257 @@
+            self.selectedComparison = attributeComparisonOperators[newIndex]
+            self.navigationController?.popViewController(animated: true)
+        }
+        optionsViewController.title = "Comparison"
+        show(optionsViewController, sender: self)
+    }
+    
+    // Transition to the value options view controller.
+    func showValuePicker(values: [AGSCodedValue]) {
+        let selectedIndex = -1
+        let valueLabels = values.map { $0.name }
+        let optionsViewController = OptionsTableViewController(labels: valueLabels, selectedIndex: selectedIndex) { newIndex in
+            self.selectedValue = values[newIndex]
+            self.valueCell.detailTextLabel?.text = valueLabels[newIndex]
+            self.navigationController?.popViewController(animated: true)
+        }
+        optionsViewController.title = "Value"
+        show(optionsViewController, sender: self)
+    }
+    
+    // Prompt an alert to allow the user to input custom values.
+    func showValueInputField(completion: @escaping (NSNumber?) -> Void) {
+        // Create an object to observe if text field input is empty.
+        var textFieldObserver: NSObjectProtocol!
+        
+        let alertController = UIAlertController(title: "Provide a comparison value", message: nil, preferredStyle: .alert)
+        let cancelAction = UIAlertAction(title: "Cancel", style: .cancel) { _ in
+            // Remove observer when canceled.
+            NotificationCenter.default.removeObserver(textFieldObserver!)
+        }
+        let doneAction = UIAlertAction(title: "Done", style: .default) { _ in
+            let textField = alertController.textFields!.first!
+            // Remove the observer when done button is no longer in use.
+            NotificationCenter.default.removeObserver(textFieldObserver!)
+            // Convert the string to a number.
+            completion(NumberFormatter().number(from: textField.text!))
+        }
+        // Add the done action to the alert controller.
+        doneAction.isEnabled = false
+        alertController.addAction(doneAction)
+        // Add a text field to the alert controller.
+        alertController.addTextField { textField in
+            textField.keyboardType = .numbersAndPunctuation
+            textField.placeholder = "e.g. 15"
+        }
+        // Add an observer to ensure the user does not input an empty string.
+        textFieldObserver = NotificationCenter.default.addObserver(
+            forName: UITextField.textDidChangeNotification,
+            object: nil,
+            queue: .main
+        ) { _ in
+            if let text = alertController.textFields!.first!.text {
+                // Enable the done button if the textfield is not empty and is a valid number.
+                doneAction.isEnabled = !text.isEmpty && NumberFormatter().number(from: text) != nil
```

Comment:
Actually after more thoughts, I think it doesn't need to check string isEmpty - empty string cannot convert to a number.